src/csharp/Microsoft.Spark/Interop/Ipc/JvmThreadPool.cs

```diff
@@ -0,0 +1,83 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Threading;
+
+namespace Microsoft.Spark.Interop.Ipc
+{
+    /// <summary>
+    /// This class corresponds to the ThreadPool we maintain on the JVM side. This class keeps
+    /// track of which .NET threads are still alive, and issues an rmThread command if a thread is
+    /// not.
```

Comment:
Can you add a comment that one .NET thread maps to one (always same) JVM thread?

Reply:
Also, comment on the thread-safe methods.

Reply:
Added more documentation and explained that all method are thread-safe.